test/e2e/annotations/affinity.go

```diff
@@ -223,4 +223,65 @@ var _ = framework.IngressNginxDescribe("Annotations - Affinity/Sticky Sessions",
 		Expect(resp.Header.Get("Set-Cookie")).Should(ContainSubstring(fmt.Sprintf("Expires=%s", expected)))
 		Expect(resp.Header.Get("Set-Cookie")).Should(ContainSubstring("Max-Age=259200"))
 	})
+
+	It("should work with use-regex annotation and session-cookie-path", func() {
+		host := "cookie.foo.com"
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/affinity":            "cookie",
+			"nginx.ingress.kubernetes.io/session-cookie-name": "SERVERID",
+			"nginx.ingress.kubernetes.io/use-regex":           "true",
+			"nginx.ingress.kubernetes.io/session-cookie-path": "/foo/bar",
+		}
+
+		ing := framework.NewSingleIngress(host, "/foo/.*", host, f.IngressController.Namespace, "http-svc", 80, &annotations)
+		f.EnsureIngress(ing)
+
+		f.WaitForNginxServer(host,
+			func(server string) bool {
+				return strings.Contains(server, fmt.Sprintf("server_name %s ;", host))
+			})
+
+		resp, _, errs := gorequest.New().
+			Get(f.IngressController.HTTPURL+"/foo/bar").
```

Comment:
Can you test this with another path i.e `foo/baz`? I think what's gonna happen now is it will still set cookie and it's Path will be `foo/bar` but I'm not sure if this is a correct behaviour. While it's probably harmless but does not make sense.

Reply:
This is exactly what happens currently however this will be fixed once this PR is merged: https://github.com/kubernetes/ingress-nginx/pull/3383

Reply:
That PR above adds a test case for what you're mentioning above so I don't think it should be added here otherwise e2e-tests will break on this branch.